src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiBehaviorApiDescriptionProvider.cs

```diff
@@ -42,6 +43,40 @@ public void OnProvidersExecuting(ApiDescriptionProviderContext context)
                 {
                     description.SupportedResponseTypes.Add(responseType);
                 }
+
+                AddRequestTypeForFormFiles(description);
+            }
+        }
+
+        private void AddRequestTypeForFormFiles(ApiDescription description)
+        {
+            if (description.SupportedRequestFormats.Count > 0)
+            {
+                // If there are already supported formats registered, don't do anything.
+                return;
+            }
+
+            if (description.ParameterDescriptions.Any(p => p.Source == BindingSource.FormFile))
+            {
+                var requestMetadataAttributes = DefaultApiDescriptionProvider.GetRequestMetadataAttributes(description.ActionDescriptor);
+                // Walk through all 'filter' attributes in order, and allow each one to see or override
+                // the results of the previous ones. This is similar to the execution path for content-negotiation.
+                var contentTypes = new MediaTypeCollection();
+                if (requestMetadataAttributes != null)
+                {
+                    foreach (var metadataAttribute in requestMetadataAttributes)
+                    {
+                        metadataAttribute.SetContentTypes(contentTypes);
+                    }
+                }
+
+                foreach (var contentType in contentTypes)
```

Comment:
Can you add a note here that we're ignoring formatters, because we don't use a formatter to read files.